src/kernels/jupyter/types.ts

```diff
@@ -202,10 +202,21 @@ export interface IJupyterServerUri {
 export type JupyterServerUriHandle = string;
 
 export interface IJupyterUriProvider {
-    readonly id: string; // Should be a unique string (like a guid)
-    getQuickPickEntryItems(): QuickPickItem[];
-    handleQuickPick(item: QuickPickItem, backEnabled: boolean): Promise<JupyterServerUriHandle | 'back' | undefined>;
+    /**
+     * Should be a unique string (like a guid)
+     */
+    readonly id: string;
+    onDidChangeHandlers?: Event<void>;
+    getQuickPickEntryItems?(): QuickPickItem[];
+    handleQuickPick?(item: QuickPickItem, backEnabled: boolean): Promise<JupyterServerUriHandle | 'back' | undefined>;
```

Comment:
Extensions such as github or others won't need to implement `getQuickPickEntryItems`
Haven't hooked up the code to make use of this API just yet, but added this nevertheless, basically as defined here https://github.com/microsoft/vscode/issues/146942

Reply:
The code that uses this has a few more changes, figured i'd submit this PR without those changes as its pretty large as is.